core/src/main/java/org/apache/iceberg/view/ViewProperties.java

```diff
@@ -26,6 +26,8 @@ public class ViewProperties {
   public static final String METADATA_COMPRESSION = "write.metadata.compression-codec";
   public static final String METADATA_COMPRESSION_DEFAULT = "gzip";
   public static final String COMMENT = "comment";
+  public static final String REPLACE_STRICTNESS_ENABLED = "replace.strictness.enabled";
```

Comment:
I think this is over-broad. There could be many forms of "strict" replacement. For example, it is more strict to ensure that the column names and positions don't change in a replace operation, but that's not what we're addressing here.

How about something like `replace.drop-dialect.allowed=false`? That would prevent the caller from dropping a view dialect.

Reply:
I agree that the inital naming was quite broad. I've updated this to `replace.drop-dialect.allowed` with `false` being the default